droidmediacamera.cpp

```diff
@@ -333,8 +339,18 @@ DroidMediaCamera *droid_media_camera_connect(int camera_number)
     }
 
     cam->m_queue = queue;
+    cam->m_queue->attachToCameraPreview(cam->m_camera);
 
-    cam->m_queue->attachToCamera(cam->m_camera);
+#if ANDROID_MAJOR >= 8
+    android::sp<DroidMediaBufferQueue>
+      video_queue(new DroidMediaBufferQueue("DroidMediaCameraBufferVideoQueue"));
+    if (!video_queue->connectListener()) {
+        ALOGE("Failed to connect video buffer queue listener");
+    } else {
+      cam->m_video_queue = video_queue;
+      cam->m_video_queue->attachToCameraVideo(cam->m_camera);
```

Comment:
I had a little bit of difficulty with video recording failing to start with my test app (https://gist.github.com/adenexter/89375f20793fcee698d623597b84ff0e) initially. The problem turned out to be that the video mode was incorrect, possibly related to me calling droid_media_camera_store_meta_data_in_buffers(), and all combinations failed; true, false, and not called at all which might be a retained setting across uses.

I fixed this for myself by adding 
```
      cam->m_camera->setVideoBufferMode(
                      android::hardware::ICamera::VIDEO_BUFFER_MODE_BUFFER_QUEUE);
```
here and not calling droid_media_camera_store_meta_data_in_buffers().

That should have been the same as calling droid_media_camera_store_meta_data_in_buffers(camera, true), but that falls back to other modes if it thinks setting the mode failed, so I think setting the mode to VIDEO_BUFFER_MODE_BUFFER_QUEUE is working but returning a status other than OK anyway.

But since the other modes won't work, maybe the mode should be set here and droid_media_camera_store_meta_data_in_buffers() ignored.

Reply:
Done. Yes, it's better that way.